strapi/strapi
# chore(content-releases): optimistic updates for content releases actions

# Current problem

![Screenshot 2023-12-15 at 11 18 44](https://github.com/strapi/strapi/assets/23050694/2f947320-da21-4705-a32b-d0f8d468465b)

In the image above we have inside the red box the ReleaseAction type, users can change this. Right now we have a little delay when users clicks this to change the type of ReleaseAction, so we want to add optimistic update to this request. 

## Main problem with the implementation

As mention [here](https://redux-toolkit.js.org/rtk-query/comparison#no-normalized-or-deduplicated-cache), RTK Query doesn't handle a normalised cache between multiple requests, actually we are caching each request by its params. 

Considering we are using pagination, this means that our cache looks like this:
```
'getReleasesActions({ releaseId: 1, page: 1, pageSize: 10 })': { ... result ... }
```

This works great for caching our results, but the problem is: How can I update the cache when making an update to the ReleaseAction? considering that in updateReleaseAction I don't have the query params for pagination.

The solution I'm trying in this PR is exactly that: I added the query params to the updateReleaseAction, in that way we can recreate the same params I use for getReleasesActions

# How to test it

You should have an EE license and the feature enabled by setting contentReleases as enabled on the feature file.
```
// myproject/config/features.js
module.exports = ({ env }) => ({
  future: {
    contentReleases: true
  },
});
```

1. Create a release and add entries to it.
2. Go to the release details page and start changing the type of any ReleaseAction, in the UI the change should be instantaneous.

**Tip:** Use the network tab to simulate a Slow 3G network so it's more obvious
